api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -440,7 +440,7 @@ public Builder year(String sourceName, String targetName) {
               sourceColumn.fieldId(),
               nextFieldId(),
               targetName,
-              Transforms.year(sourceColumn.type()));
```

Comment:
I'm hesitant to remove those because then there is a situation where we don't know the `sourceColumn`.